# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1711492897 -3600
# Parent  f029df86d0abce20d4e51ad6078b7fd47d96fc79
Misc: Don't show add-on 'last updated' dates before 2000.

Fix for https://github.com/flathub/eu.betterbird.Betterbird/issues/126. The FlatPak language packs are distribution extensions
which for some reason have a 'last updated' date of 01 Jan 1970. Anything before the year 2000 is impossible anyway.

diff --git a/toolkit/mozapps/extensions/content/aboutaddons.js b/toolkit/mozapps/extensions/content/aboutaddons.js
--- a/toolkit/mozapps/extensions/content/aboutaddons.js
+++ b/toolkit/mozapps/extensions/content/aboutaddons.js
@@ -2323,9 +2323,9 @@ class AddonDetails extends HTMLElement {
     }
 
     // Last updated.
     let updateDate = this.querySelector(".addon-detail-row-lastUpdated");
-    if (addon.updateDate) {
+    if (addon.updateDate && addon.updateDate.getFullYear() >= 2000) {
       let lastUpdated = addon.updateDate.toLocaleDateString(undefined, {
         year: "numeric",
         month: "long",
         day: "numeric",
